temporal-sdk/src/main/java/io/temporal/client/WorkflowClientInternalImpl.java

```diff
@@ -102,6 +109,21 @@ public static WorkflowClient newInstance(
             options.getDataConverter());
   }
 
+  private List<WorkflowClientInterceptor> initializeInterceptors() {
+    List<WorkflowClientInterceptor> interceptors =
+        new ArrayList<>(options.getInterceptors().length + 1);
+    interceptors.add(
+        new WorkflowClientInterceptorBase() {
+          @Override
+          public WorkflowClientCallsInterceptor workflowClientCallsInterceptor(
+              WorkflowClientCallsInterceptor next) {
+            return new LocalDispatchWorkflowClientInterceptor(next, workerFactories);
```

Comment:
Can be and it's much easier to do. Originally I did it this way, but after that I realized it should be better as an interceptor:
1. It allows user to implement their custom disabling logic
2. It provides an access for any kind of reporting interceptor user may want to implement in the interceptors.
3. can be an interceptor which helps code modularity. This keeps implementation highly cohesive and better structured.
4. Should be an interceptor, because there is no significant shortfalls while we are getting ^